backend/sftp/file.go

```diff
@@ -260,11 +265,13 @@ func (f *File) Read(p []byte) (n int, err error) {
 	f.fileSystem.connTimerStop()
 	defer f.fileSystem.connTimerStart()
 
-	sftpfile, err := f.openFile(os.O_RDONLY)
+	sftpfile, err := f.openFile(os.O_RDWR)
```

Comment:
This allows for (contrived but possible) sequence of reading some bytes in a file (for instance the first line (like csv header), then overwriting things after that.   Readonly would've discounted the possiblity or writing after read.